pkg/channel/consolidated/status/status.go

```diff
@@ -317,6 +320,11 @@ func (m *Prober) Start(done <-chan struct{}) chan struct{} {
 func (m *Prober) CancelProbing(sub eventingduckv1.SubscriberSpec) {
 	m.mu.Lock()
 	defer m.mu.Unlock()
+	m.forgetState(sub)
+}
+
+// This is meant to be called while m.mu is locked
+func (m *Prober) forgetState(sub eventingduckv1.SubscriberSpec) {
```

Comment:
could you elaborate to me why it's a sound design decision to have:
`CancelProbing` (which is part of the public Manager interface)
`CancelProbing`'s single job is to aquire the mutex lock, and call `forgetState(sub)`
`forgetState(sub)` has a comment a comment that indicates that we *must* have the lock before calling it
we then have other ways of calling `forgetState` elsewhere in the file.

Why aren't we using `CancelProbing` as the proper entrypoint my consistently?

Reply:
OK, so the issue is:

for outside invocation he have the `CancelProbe`, that basically does the lock and calls the private one.

Internally we call the `forget` since we have already taken care of the lock.

Perhaps the different names of `CancelProbe` versus `froget...` is confusing?

I think I understand the way of decorating the invocation of the `forget` with a lock, inside of the `CancelProbe`.

So...  Let's perhaps change the name?  `internalCancelProbe`, and be more explicit that it's only used w/in a mutex, and outside users should be instead calling `CancelProbe` ? 

And perhaps add a TODO... to have that sorted later